system/jlib/jstring.cpp

```diff
@@ -2482,3 +2501,49 @@ const char * nullText(const char * text)
     if (text) return text;
     return "(null)";
 }
+
+StringBuffer& StringBuffer::operator=(StringBuffer&& value)
+{
+    swapWith(value);
+    return *this;
+}
+
+StringBuffer operator+ (const StringBuffer& lhs, const StringBuffer& rhs)
```

Comment:
In particular for StringBuffer the object is relatively large, and the operator overloading is hiding a large degree of inefficiency.  It is much more efficient to declare a StringBuffer and then append to it:

StringBuffer x;
x.append(a).append(b).append(c);